### PR TITLE
Add Function retry when the worker has not been fully initialized

### PR DESCRIPTION
**Pull request by sophiatev**
# Summary
## What changed?
This PR adds a new case for Activity/orchestration/entity execution failures that detects errors related to a worker not being fully initialized, and throws a `SessionAbortedException` in these cases which will ensure the Function execution is retried rather than failed.

## Why is this change needed?
Previously these Function invocations were marked as failed, but they should be retried for platform-related errors.

## Issues / work items
- Resolves #3438 

---

# Project checklist
- [ ] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [ ] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [ ] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [ ] All required tests have been added/updated (unit tests, E2E tests)
- [ ] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [ ] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [ ] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [ ] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s):
- AI-assisted areas/files:
- What you changed after AI output:

AI verification (required if AI was used):
- [ ] I understand the code and can explain it
- [ ] I verified referenced APIs/types exist and are correct
- [ ] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [ ] I reviewed concurrency/async behavior
- [ ] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed / Failed (link logs if failed)

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components):
- Steps + observed results:
  1.
  2.
  3.
- Evidence (optional):

---

# Notes for reviewers
- N/A

### CODE REVIEW

**Comment by sophiatev** [comment]
@copilot Update `OutOfProcMiddlewareTests` to add tests for the new cases in this PR

**Comment by sophiatev** [comment]
@copilot Add a test for the entity case too now

File: src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs [hunk 1]
```diff
@@ -666,7 +673,14 @@ private static bool IsPlatformLevelException(Exception? exception)
...
                 || (exception?.InnerException is InvalidOperationException ioe
-                    && ioe.Message.Contains(NoProcessAssociatedMessage, StringComparison.Ordinal));
+                    && ioe.Message.Contains(NoProcessAssociatedMessage, StringComparison.Ordinal))
+                || IsWorkerNotFullyInitializedException(exception);
+        }
+
+        private static bool IsWorkerNotFullyInitializedException(Exception? exception)
+        {
+            return (exception?.InnerException is InvalidOperationException ioe && ioe.Message.Contains(NoWorkerInitializedMessage, StringComparison.Ordinal))
+                || (exception?.InnerException is FileNotFoundException fnfe && fnfe.Message.Contains(AssemblyNotLoadedMessage, StringComparison.Ordinal));
```
Comment (AnatoliB) [inline src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs:683]: I'm concerned about the "assembly not loaded" exception being potentially thrown by customer code for reasons unrelated to the issue we are fixing. Is there anything else that we can use to distinguish between these two cases? Anything different on the stack trace?

Comment (sophiatev) [reply src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs:683]: This is a good point - I think the outer exception would be of type `FunctionInvocationException` in this case, although that may be the case even for customer-code-errors... let me see if I can find anything useful for this

Comment (sophiatev) [reply src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs:683]: I had copilot do a small test for me where it threw a `FileNotFoundException` from user code and this is what it found:

"Your inner-type checking is the right discriminator — `FileNotFoundException` and `InvalidOperationException` instances only appear there in the host-dispatch path; user-code exceptions of those types would arrive as an RpcException with "Result: ..." text and would not match."

In other words, when the exception originates from user code, the inner exception type is actually `RpcException`. This is already something we rely on in our code in the same class [here](https://github.com/Azure/azure-functions-durable-extension/blob/f0c960b01e57e893b014fed22cdd431ad4d2d82a/src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs#L672)  when attempting to extract failure details related to user exceptions. So I think the check we have now should distinguish between the two cases already.

Comment (AnatoliB) [reply src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs:683]: Ok, this what I was hoping for, thank you for checking!

**Review by AnatoliB — APPROVED** [review]
